internal/fields/validate.go

```diff
@@ -0,0 +1,196 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fields
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/common"
+)
+
+// Validator is responsible for fields validation.
+type Validator struct {
+	schema []fieldDefinition
+}
+
+// CreateValidatorForDataStream method creates a validator for the data stream.
+func CreateValidatorForDataStream(dataStreamRootPath string) (*Validator, error) {
+	fieldsDir := filepath.Join(dataStreamRootPath, "fields")
+	fis, err := ioutil.ReadDir(fieldsDir)
```

Comment:
Nit: It's not clear to me what `fis` stand for. But I understand that it will contain a list of fields files. Maybe rename the variable to `fieldsFiles` instead of `fis` to be a bit more descriptive and to parallel `fieldsDir`?

Reply:
This is actually `fileInfos`, but I agree it's not really descriptive. Renamed!